specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/publicIpAddress.json

```diff
@@ -360,6 +360,53 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/publicIPAddresses/{publicIpAddressName}/ddosProtectionStatus": {
```

Comment:
>ddosProtectionStatus [](http://example.com/codeflow?start=140&length=20)

any reason these are modelled as POST operaitons ? .. 

Both these operation on publicIpaddress and the virtual network seem like they should be modelled as GET operations on a resource instead of the POST which should typically be used for actions on resoures

Reply:
Yep - so the reason why these are POST operations is that these APIs call an external service (SLB) to get the effective ddos protection status of a public IP or VNet. Because GET in NRP has a stricter performance limit/bar - we were advised from NRP folks that we need to use POST for these 2 APIs as we're fetching this information outside of NRP, mainly due to the fact that this information is from an external service and not directly from NRP.

Reply:
It is higly recommended in ARM to have these modelled as a GET especially since clients including the UI , are familiar with a collection get and work well with things like skiptoken and top which you have in the other api. However if this is a hard requirement from NRP side it is fine. Signing off. Please let me know if you need any further discussion points.